app/services/tokenBalanceService.ts

```diff
@@ -0,0 +1,217 @@
+      `page=${page}`,
+      `per_page=${batchSize}`,
+      `view=scores_minimal`,
+    ].join("&");
+
+    const res = await fetch(`${baseUrl}?${queryString}`, {
+      headers: {
+        Accept: "application/json",
+        "X-API-KEY": apiKey,
+      },
+    });
+
+    if (!res.ok) {
+      const errorText = await res.text();
+      console.error(`API Error for page ${page}:`, errorText);
+      throw new Error(`Failed to fetch page ${page}: ${errorText}`);
+    }
+
+    const json = await res.json();
+    const profiles = json.profiles || [];
+    allProfiles = [...allProfiles, ...profiles];
+
+    // Break if we got fewer results than requested
+    if (profiles.length < batchSize) break;
+
+    // Add a small delay between requests
+    if (page < Math.ceil(200 / batchSize)) {
+      await new Promise((resolve) => setTimeout(resolve, 100));
+    }
+  }
+
+  return allProfiles.slice(0, 200);
+}
+
+/**
+ * Get cached token balances with 24-hour TTL
+ * FIXED: Proper cache key strategy for development mode
+ */
+export const getCachedTokenBalances = unstable_cache(
+  async (apiKey: string): Promise<TokenBalanceCache> => {
+    console.log("🔄 Fetching fresh token balances for all top 200 creators...");
+
+    try {
+      // Fetch top 200 profiles
+      const profiles = await fetchTop200Profiles(apiKey);
+      const tokenBalances: Record<string, TokenBalanceData> = {};
+
+      // Fetch token balances for all profiles in parallel (with rate limiting)
+      const batchSize = 10;
+      for (let i = 0; i < profiles.length; i += batchSize) {
+        const batch = profiles.slice(i, i + batchSize);
+        const batchPromises = batch.map((profile) =>
+          getTokenBalanceForProfile(profile.id, apiKey),
+        );
+
+        const batchResults = await Promise.all(batchPromises);
+
+        batch.forEach((profile, index) => {
+          tokenBalances[profile.id] = batchResults[index];
+        });
+
+        // Add delay between batches to be nice to the API
+        if (i + batchSize < profiles.length) {
+          await new Promise((resolve) => setTimeout(resolve, 200));
+        }
+      }
+
+      const lastUpdated = new Date().toISOString();
+      const nextUpdate = new Date(
+        Date.now() + 6 * 60 * 60 * 1000,
+      ).toISOString(); // 6 hours from now
+
+      const boostedCount = Object.values(tokenBalances).filter(
+        (data) => data.isBoosted,
+      ).length;
+
+      console.log(
+        `✅ Token balance cache updated: ${Object.keys(tokenBalances).length} profiles, ${boostedCount} boosted`,
+      );
+
+      return {
+        tokenBalances,
+        lastUpdated,
+        nextUpdate,
+      };
+    } catch (error) {
+      console.error("❌ Error updating token balance cache:", error);
+      throw error;
+    }
+  },
+  ["token-balances-v2"], // FIXED: Unique cache key to avoid conflicts
+  {
+    revalidate: 24 * 60 * 60, // 24 hours TTL
+    tags: ["token-balances"], // FIXED: Add tags for cache invalidation
```

Comment:
we have a cache file to define this so it's easier to manage non-duplicated keys. Can you add both the 24 hours and the key you are using for the cache please?